haas/ext/network_allocators/vlan_pool.py

```diff
@@ -0,0 +1,82 @@
+"""VLAN based ``network_allocator`` implementation."""
+
+import logging
+
+from haas.network_allocator import NetworkAllocator, set_network_allocator
+from haas.model import AnonModel
+from haas.config import cfg
+from sqlalchemy import Column, Integer, Boolean
+
+
+def get_vlan_list():
+    """Return a list of vlans in the module's config section.
+
+    This is for use by the ``create_bridges`` script.
+    """
+    vlan_str = cfg.get(__name__, 'vlans')
+    returnee = []
+    for r in vlan_str.split(","):
+        r = r.strip().split("-")
+        if len(r) == 1:
+            returnee.append(int(r[0]))
+        else:
+            returnee += range(int(r[0]), int(r[1])+1)
+    return returnee
+
+
+class VlanAllocator(NetworkAllocator):
+    """A allocator of VLANs. The interface is as specified in ``NetworkAllocator``."""
+
+    def get_new_network_id(self, db):
+        vlan = db.query(Vlan).filter_by(available=True).first()
+        if not vlan:
+            return None
```

Comment:
Stylistic (take or leave): is it better to return `None` or throw an exception in this case? Could there ever be a `get_new_network_id()` that legitimately returned `None`?